src/Resources/contao/classes/FrontendTemplate.php

```diff
@@ -250,12 +250,9 @@ public static function addToUrl($strRequest, $blnIgnoreParams=false, $arrUnset=a
 	 */
 	public function hasAuthenticatedBackendUser()
 	{
-		if (!isset($_COOKIE['BE_USER_AUTH']))
-		{
-			return false;
-		}
+		$objTokenChecker = \System::getContainer()->get('contao.security.token_checker');
```

Comment:
Why is this accessing the global `System` and the global `BackendUser`. I know it's in old code but we don't use these anymore.

Reply:
Again, this is a "do not fix just one of 1,000 occurrences" case. 😄

Reply:
We are trying to keep the legacy code untouched, so it is easier to synchronize `contao/core-bundle` with `contao/core`. After June 2019, we can change all occurrences.

Reply:
Because it's legacy code …